calico_rkt/calico_rkt.py

```diff
@@ -1,91 +1,249 @@
+from pycalico.datastore import IF_PREFIX, DatastoreClient
+from pycalico.datastore_errors import PoolNotFound
+
+print_stderr = functools.partial(print, file=sys.stderr)
+
+
+input_ = ''.join(sys.stdin.readlines()).replace('\n', '')
+input_json = json.loads(input_)
+
+# Append to existing env, to avoid losing PATH etc.
+# TODO-PAT: This shouldn't be hardcoded
+env = os.environ.copy()
+ETCD_AUTHORITY_ENV = 'ETCD_AUTHORITY'
+env[ETCD_AUTHORITY_ENV] = 'localhost:2379'
+
+# PROFILE_LABEL = 'CALICO_PROFILE'
+# ETCD_PROFILE_PATH = '/calico/'
+
+ORCHESTRATOR_ID = "rkt"
+HOSTNAME = socket.gethostname()
+NETNS_ROOT= '/var/lib/rkt/pods/run'
 
 def main():
-    stdin_raw_data = sys.stdin.read()
+    mode = env['CNI_COMMAND']
+
+    if mode == 'init':
+        print_stderr('No initialization work to perform')
+    elif mode == 'ADD':
+        print_stderr('Executing Calico pod-creation plugin')
+        create(
+            container_id=env['CNI_CONTAINERID']
+            )
+    elif mode == 'DEL':
+        print_stderr('Executing Calico pod-deletion plugin')
+        delete(
+            container_id=env['CNI_CONTAINERID']
+            )
+
+def create(container_id):
+    """"Handle rkt pod-create event."""
+    print_stderr('Configuring pod %s' % container_id, file=sys.stderr)
+    netns_path='%s/%s/%s' % (NETNS_ROOT, container_id, env['CNI_NETNS'])
+    _datastore_client = IPAMClient()
 
-    # Convert input data to JSON object
     try:
-        stdin_json = json.loads(stdin_raw_data)
-    except ValueError as e:
-        quit_with_error(str(e))
-
-    # Extract command
+        endpoint, ip = _create_calico_endpoint(container_id=container_id,
+                                            netns_path=netns_path,
+                                            client=_datastore_client)
+
+        _create_profile(endpoint=endpoint, 
+                        profile_name=input_json['name'], 
+                        ip=ip,
+                        client=_datastore_client)
+    except CalledProcessError as e:
+        print_stderr('Error code %d creating pod networking: %s\n%s' % (
+            e.returncode, e.output, e))
+        sys.exit(1)
+    print_stderr('Finished Creating pod %s' % container_id)    
+
+def delete(container_id):
+    """Cleanup after a pod."""
+    print_stderr('Deleting pod %s' % container_id, file=sys.stderr)
+
+    _datastore_client = IPAMClient()
+
+    # Remove the profile for the workload.
+    _container_remove(hostname=HOSTNAME,
+                        orchestrator_id=ORCHESTRATOR_ID,
+                        container_id=container_id,
+                        client=_datastore_client)
+
+    # Delete profile
     try:
-        command = stdin_json['command']
-    except KeyError:
-        quit_with_error("Missing command")
-
-    # Extract args
+        _datastore_client.remove_profile(input_json['name'])
+    except:
+        print_stderr("Cannot remove profile %s; Profile cannot be found." % container_id)
+
```

Comment:
You should have a print at the bottom here for then the pod has been removed.  Make sure you add a `sys.exit` call above too when you do this!